clients/src/main/java/org/apache/kafka/clients/consumer/internals/LegacyKafkaConsumer.java

```diff
@@ -150,11 +150,9 @@ public class LegacyKafkaConsumer<K, V> implements ConsumerDelegate<K, V> {
             LogContext logContext = createLogContext(config, groupRebalanceConfig);
             this.log = logContext.logger(getClass());
             boolean enableAutoCommit = config.getBoolean(ENABLE_AUTO_COMMIT_CONFIG);
-            groupId.ifPresent(groupIdStr -> {
-                if (groupIdStr.isEmpty()) {
-                    log.warn("Support for using the empty group id by consumers is deprecated and will be removed in the next major release.");
-                }
-            });
+
+            if (this.groupId.isPresent() && this.groupId.get().isEmpty())
+                throw new InvalidGroupIdException("The configured group.id should not be an empty string or whitespace");
```

Comment:
I don't think we can make this change in a minor release, but we need to wait for 4.0. It's break compatibility, and we are not allowed to break it in a minor release.

Reply:
Fair enough. I'll back out the error case for `LegacyKafkaConsumer`.